rama-http-backend/src/client/mod.rs

```diff
@@ -440,103 +158,404 @@ where
     }
 }
 
-#[cfg(feature = "boring")]
-fn create_connector_data_boring<State>(
-    ctx: &Context<State>,
-    tls_config: &Option<ClientConfig>,
-) -> Result<Arc<BoringTlsConnectorDataBuilder>, OpaqueError> {
-    let builder = match extract_client_config_from_ctx(ctx) {
-        Some(mut chain_ref) => {
-            trace!(
-                "create tls connector using rama tls client config(s) from context and/or the predefined one if defined"
-            );
-            if let Some(tls_config) = tls_config {
-                chain_ref.prepend(tls_config.clone());
+#[doc(inline)]
+pub use easy_connector::EasyHttpWebClientBuilder;
+
+mod easy_connector {
+    use super::{HttpConnector, proxy::layer::HttpProxyConnector};
+    use rama_core::{
+        Service,
+        error::{BoxError, OpaqueError},
+    };
+    use rama_dns::DnsResolver;
+    use rama_http::{Request, dep::http_body};
+    use rama_net::client::{
+        EstablishedClientConnection,
+        pool::{
+            FiFoReuseLruDropPool, PooledConnector,
+            http::{
+                BasicHttpConId, BasicHttpConnIdentifier, HttpPooledConnectorBuilder,
+                HttpPooledConnectorConfig,
+            },
+        },
+    };
+    use rama_tcp::client::service::TcpConnector;
+    use std::{marker::PhantomData, time::Duration};
+
+    #[cfg(feature = "boring")]
+    use ::{rama_tls_boring::client as boring_client, std::sync::Arc};
+
+    #[cfg(feature = "rustls")]
+    use rama_tls_rustls::client as rustls_client;
+
+    #[cfg(any(feature = "rustls", feature = "boring"))]
+    use super::http_inspector::HttpsAlpnModifier;
+
+    /// Builder that is designed to easily create a [`super::EasyHttpWebClient`] from most basic use cases
+    pub struct EasyHttpWebClientBuilder<C, S> {
+        connector: C,
+        _phantom: PhantomData<S>,
+    }
+
+    pub struct TransportStage;
+    pub struct ProxyTunnelStage;
+    pub struct ProxyStage;
+    pub struct HttpStage;
+    pub struct PoolStage;
+
+    pub(super) type NewBuilder = EasyHttpWebClientBuilder<TcpConnector, TransportStage>;
+
+    impl EasyHttpWebClientBuilder<(), ()> {
+        pub fn new() -> NewBuilder {
+            EasyHttpWebClientBuilder {
+                connector: TcpConnector::new(),
+                _phantom: PhantomData,
+            }
+        }
+    }
+
+    impl EasyHttpWebClientBuilder<TcpConnector, TransportStage> {
+        /// Add a custom [`DnsResolver`] that will be used by this client
+        pub fn with_dns_resolver<T: DnsResolver + Clone>(
```

Comment:
FYI in future we'll also have a dns resolver for the tls connector (e.g. to support fetching pre-knowledge records).; Question will be if we need to therefore see these as 2 separate resolvers, or if we would just move the dns resolver to the client builder level.

You can close this as-is, just letting you know so it can sink in a bit.